src/ffjord.jl

```diff
@@ -169,7 +188,12 @@ _trace_batched(x::AbstractArray{T, 3}) where T =
 
 function ffjord(u, p, t, re, e;
                 regularize=false, monte_carlo=true)
-    m = re(p)
+    # m = re(p)
+    if isnothing(re)
+        m = p
```

Comment:
FFJORD still needs some work because here m will be a `ComponentVector` which is not callable when doing  `mz = m(z)` which throws an error. Not sure how to resolve this.

Reply:
This isn't correct. `m` is the model, not the result of the model. Flux `re(p)(x)` is equal to Lux `model(x,p,st)`. You should dispatch on whether `Flux.destructure` is done by making one dispatch for `re::Lux.AbstractExplicitLayer`, i.e. it should be:

```julia
function ffjord(u, p, t, model::AbstractExplicitLayer, e, st;
                regularize=false, monte_carlo=true)

function ffjord(u, p, t, re, e;
                regularize=false, monte_carlo=true)
```

for the Lux vs Flux cases.